### PR TITLE
Add OpenApi annotations for /infra/{}/errors

### PR DESCRIPTION
**Pull request by leovalais**
refs #4580 #3625

related issue #6349

### CODE REVIEW

**Review by younesschrifi — APPROVED** [review]
LGTM (Editoast part)

File: editoast/src/schema/errors.rs [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
-#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+crate::schemas! {
+    InfraError,
+    InfraErrorType,
+    InfraErrorTypeLabel,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone, ToSchema)]
+// FIXME: this stuct do not deserialize properly because deny_unknown_fields
+// do not work with flatten. Cf. https://github.com/osrd-project/osrd/issues/6349
```
Comment (flomonster) [inline editoast/src/schema/errors.rs:15]: We could remove the `deny_unknown_fields` to simplify what we're trying to do.

**Review by flomonster — COMMENTED** [review]
Reviewed but not tested yet.

**Comment by leovalais** [comment]
Well I guess that re-doing it would be less work than solving that conflict now 😞

**Pull request closed without merging**